File: apps/mobile/android/app/src/main/java/br/com/aipinho/mobile/network/GeminiExecutorClient.kt
package br.com.aipinho.mobile.network

import br.com.aipinho.mobile.models.ConnectionProfile
import br.com.aipinho.mobile.ui.policies.ChatAutoRefreshPolicy

class GeminiExecutorClient(profile: ConnectionProfile, token: () -> String?) : BaseApiClient(profile, profile.corePort, token, ChatAutoRefreshPolicy.chatRequestTimeoutMs) {
    fun health() = get("/api/v1/gemini-executor/health")
    fun configStatus() = get("/api/v1/gemini-executor/config/status")
    fun sessions() = get("/api/v1/gemini-executor/sessions")
    fun createSession() = post("/api/v1/gemini-executor/sessions", "{\"title\":\"Gemini Executor\"}")
    fun renameSession(sessionId: String, title: String) = post("/api/v1/gemini-executor/sessions/$sessionId/rename", "{\"title\":${JsonPayload.string(title)}}")
    fun deleteSession(sessionId: String) = delete("/api/v1/gemini-executor/sessions/$sessionId")
    fun messages(sessionId: String) = get("/api/v1/gemini-executor/sessions/$sessionId/messages")
    fun send(sessionId: String, prompt: String, workspace: String = "") = post(
        "/api/v1/gemini-executor/sessions/$sessionId/send",
        payload(sessionId, prompt, workspace, "gemini_chat", emptyList())
    )
    fun plan(sessionId: String, prompt: String, workspace: String = "") = post(
        "/api/v1/gemini-executor/sessions/$sessionId/plan",
        payload(sessionId, prompt, workspace, "gemini_coding_plan", if (workspace.isBlank()) emptyList() else listOf("read_workspace", "scan_workspace"))
    )
    fun preview(sessionId: String, prompt: String, workspace: String = "") = post(
        "/api/v1/gemini-executor/sessions/$sessionId/preview",
        payload(sessionId, prompt, workspace, "gemini_patch_preview", listOf("read_workspace", "scan_workspace", "create_patch_preview"))
    )

    private fun payload(sessionId: String, prompt: String, workspace: String, operationType: String, capabilities: List<String>): String {
        val workspaceValue = if (workspace.isBlank()) "null" else JsonPayload.string(workspace)
        val caps = capabilities.joinToString(",") { JsonPayload.string(it) }
        return "{" +
            "\"session_id\":${JsonPayload.string(sessionId)}," +
            "\"prompt\":${JsonPayload.string(prompt)}," +
            "\"workspace_context\":$workspaceValue," +
            "\"operation_type\":${JsonPayload.string(operationType)}," +
            "\"requested_capabilities\":[$caps]" +
            "}"
    }
}

File: apps/launcher/ui/api/gemini_executor_client.py
from __future__ import annotations

from apps.launcher.ui.api.base_client import BaseClient


class GeminiExecutorClient(BaseClient):
    def health(self):
        return self.get("/api/v1/gemini-executor/health")

    def config_status(self):
        return self.get("/api/v1/gemini-executor/config/status")

    def sessions(self):
        return self.get("/api/v1/gemini-executor/sessions")

    def create_session(self, title: str = "Gemini Executor"):
        return self.post("/api/v1/gemini-executor/sessions", {"title": title})

    def messages(self, session_id: str):
        return self.get(f"/api/v1/gemini-executor/sessions/{session_id}/messages")

    def rename_session(self, session_id: str, title: str):
        return self.post(f"/api/v1/gemini-executor/sessions/{session_id}/rename", {"title": title})

    def delete_session(self, session_id: str):
        return self.delete(f"/api/v1/gemini-executor/sessions/{session_id}")

    def send(self, session_id: str, prompt: str, workspace_context: str = "", capabilities: list[str] | None = None):
        return self.post(
            f"/api/v1/gemini-executor/sessions/{session_id}/send",
            {
                "session_id": session_id,
                "prompt": prompt,
                "workspace_context": workspace_context or None,
                "operation_type": "gemini_chat",
                "requested_capabilities": capabilities or [],
            },
        )

    def plan(self, session_id: str, prompt: str, workspace_context: str = ""):
        return self.post(
            f"/api/v1/gemini-executor/sessions/{session_id}/plan",
            {
                "session_id": session_id,
                "prompt": prompt,
                "workspace_context": workspace_context or None,
                "operation_type": "gemini_coding_plan",
                "requested_capabilities": ["read_workspace", "scan_workspace"] if workspace_context else [],
            },
        )

    def preview(self, session_id: str, prompt: str, workspace_context: str = ""):
        return self.post(
            f"/api/v1/gemini-executor/sessions/{session_id}/preview",
            {
                "session_id": session_id,
                "prompt": prompt,
                "workspace_context": workspace_context or None,
                "operation_type": "gemini_patch_preview",
                "requested_capabilities": ["read_workspace", "scan_workspace", "create_patch_preview"],
            },
        )
